WalletWasabi/Services/UpdateManager.cs

```diff
@@ -140,11 +184,62 @@ private async Task<(Version Version, string DownloadUrl, string FileName)> GetLa
 			assetDownloadUrls.Add(asset["browser_download_url"]?.ToString() ?? throw new InvalidDataException("Missing download url from response."));
 		}
 
+		bool isReleaseValid = await ValidateWasabiSignatureAsync(softwareVersion, assetDownloadUrls).ConfigureAwait(false);
+		if (!isReleaseValid)
+		{
+			throw new InvalidOperationException($"Downloading new release was aborted, Wasabi signature was invalid.");
+		}
 		(string url, string fileName) = GetAssetToDownload(assetDownloadUrls);
 
 		return (githubVersion, url, fileName);
 	}
 
+	private async Task<bool> ValidateWasabiSignatureAsync(string softwareVersion, List<string> assetDownloadUrls)
+	{
+		var sha256SumsFilePath = Path.Combine(InstallerDir, "SHA256SUMS.asc");
+		var wasabiSigFilePath = Path.Combine(InstallerDir, "SHA256SUMS.wasabisig");
+
+		using HttpClient httpClient = new();
+		string sha256SumsUrl = assetDownloadUrls.Where(url => url.Contains("SHA256SUMS.asc")).First();
+		string wasabiSigUrl = assetDownloadUrls.Where(url => url.Contains("SHA256SUMS.wasabisig")).First();
+
+		try
+		{
+			using (var stream = await httpClient.GetStreamAsync(sha256SumsUrl, CancellationToken).ConfigureAwait(false))
+			{
+				await CopyStreamContentToFileAsync(stream, sha256SumsFilePath).ConfigureAwait(false);
+				Sha256SumsFilePath = sha256SumsFilePath;
+			}
+
+			using (var stream = await httpClient.GetStreamAsync(wasabiSigUrl, CancellationToken).ConfigureAwait(false))
+			{
+				await CopyStreamContentToFileAsync(stream, wasabiSigFilePath).ConfigureAwait(false);
+			}
+
+			await WasabiSignerHelpers.VerifySha256SumsFileAsync(sha256SumsFilePath).ConfigureAwait(false);
```

Comment:
This could be done "outside" completely independently. If the signature file is on github and it is not valid - then game over. If the file is not there, then you can continue with the dl.

Reply:
But the sig file should be there, we should always deliver it. In case if it's missing it could be another lead that the user downloaded a malicious software, no?